proxy/mux/mux_test.go

```diff
@@ -452,26 +457,33 @@ func (s *ServerSuite) TestSNI(c *C) {
 	b := MakeBatch(Batch{
 		Host:     "localhost",
 		Addr:     "localhost:41000",
-		Route:    `Host("localhost") && Path("/")`,
+		Route:    `Path("/path1")`,
 		URL:      e.URL,
 		Protocol: engine.HTTPS,
 		KeyPair:  &engine.KeyPair{Key: localhostKey, Cert: localhostCert},
 	})
 	b2 := MakeBatch(Batch{
 		Host:     "otherhost",
 		Addr:     "localhost:41000",
-		Route:    `Host("otherhost") && Path("/")`,
+		Route:    `Path("/path2")`,
 		URL:      e2.URL,
 		Protocol: engine.HTTPS,
-		KeyPair:  &engine.KeyPair{Key: localhostKey2, Cert: localhostCert2},
+		KeyPair:  &engine.KeyPair{Key: otherHostKey, Cert: otherHostCert},
 	})
 	b2.H.Settings.Default = true
 	c.Assert(s.mux.Init(MakeSnapshot(b, b2)), IsNil)
 	c.Assert(s.mux.Start(), IsNil)
 
-	// When/Then
-	c.Assert(GETResponse(c, b.FrontendURL("/"), testutils.Host("localhost")), Equals, "Hi, I'm endpoint 1")
-	c.Assert(GETResponse(c, b.FrontendURL("/"), testutils.Host("otherhost")), Equals, "Hi, I'm endpoint 2")
+	// For the same host, but different paths, return different endpoints (routing test)
+	c.Assert(GETResponse(c, b.FrontendURL("/path1"), testutils.Host("example.com")), Equals, "Hi, I'm endpoint 1")
+	c.Assert(GETResponse(c, b.FrontendURL("/path2"), testutils.Host("example.com")), Equals, "Hi, I'm endpoint 2")
+
+	//For the same path, if the Hostname is different (SNI), then return a different Cert - true host differentiation.
+	c.Assert(GETPeerCertSerialNo(c, b.FrontendURL("/path1"), testutils.Host("example.com")), Equals, "77bdc3e97d00584f03faec7cda682cf")
+	c.Assert(GETPeerCertSerialNo(c, b.FrontendURL("/path1"), testutils.Host("otherhost")), Equals, "c3244866e57c7b1f")
```

Comment:
Should not this fail because `path1` is not defined for otherhost?

Reply:
I guess the only thing that shit test should verify is that we serve correct TLS certificate based on the host. Host based routing is checked in another test. So for that we do not even need two responders. Because all we care is the cert no.